tests/misc_env/push_cosbench_workload.py

```diff
@@ -0,0 +1,207 @@
+  <storage type="s3" config="timeout=900000;accesskey=x;secretkey=y;endpoint=workload_endpoint;path_style_access=true"/>
+  <auth type="none"/>
+  <workflow>
+
+<!-- Initialization -->
+    <workstage name="init_containers">
+        <work type="init" workers="2" config="cprefix=pri-bkt;containers=r(1,6)"/>
+    </workstage>
+
+    <workstage name="preparing_cluster">
+        <work type="prepare" workers="2" config="cprefix=pri-bkt;containers=r(1,6);oprefix=pri-obj;
+        objects=r(1,objects_count);sizes=h(1|5|25,5|50|40,50|256|25,256|512|5,512|1024|3,1024|5120|1,5120|51200|1)KB"/>
+    </workstage>
+  </workflow>
+</workload>"""
+
+avail_storage = 0
+
+
+def prepare_fill_workload(ceph_cluster, client, rgw, controller, config):
+    """
+    preparing cosbench workload from the template
+
+    Args:
+        ceph_cluster:   Cluster participating in the test.
+        client: client node
+        rgw: rgw node
+        controller: controller node
+        config: config from test
+
+    Returns:
+        workload xml file name which is created on controller node
+    """
+    global fill_workload, avail_storage
+    keys = get_or_create_user(client)
+    fill_workload = fill_workload.replace(
+        "accesskey=x", f"accesskey={keys['access_key']}"
+    )
+    fill_workload = fill_workload.replace(
+        "secretkey=y", f"secretkey={keys['secret_key']}"
+    )
+
+    avail_storage = utils.calculate_available_storage(client)
+    LOG.info(f"Total available storage: {avail_storage}")
+    fill_percent = config.get("fill_percent", 30)
+    bytes_to_fill = avail_storage / 100 * fill_percent
+    LOG.info(f"no of bytes to fill {fill_percent} percent: {bytes_to_fill}")
+    # these bytes have to be filled in 6 buckets, so finding bytes per bucket
+    bytes_to_fill = bytes_to_fill / 6
+    # 404.56 KB is the average size according to sizes range in workload
+    # using the average size to find number of objects
+    objects_count = math.floor(bytes_to_fill * 100 / (40456 * 1024))
+    LOG.info(f"no of objects for an average of sizes in workload: {objects_count}")
+    fill_workload = fill_workload.replace("objects_count", f"{objects_count}")
+
+    workload_endpoint = "http://localhost:5000"
+    if not config.get("drivers"):
+        ip = rgw.ip_address
+        out, err = rgw.exec_command(
+            sudo=True, cmd="ceph orch ls --format json --service-type rgw"
+        )
+        rgw_service = json.loads(out)
+        port = rgw_service[0]["status"]["ports"][0]
+        workload_endpoint = f"http://{ip}:{port}"
+    LOG.info(f"workload endpoint: {workload_endpoint}")
+    fill_workload = fill_workload.replace("workload_endpoint", workload_endpoint)
+
+    out, err = rgw.exec_command(
+        cmd="sh /opt/cosbench/cli.sh info | grep drivers | awk '{print $2}'"
+    )
+    LOG.info(out)
+    drivers_count = int(out.strip())
+    workers = drivers_count * 100
+    fill_workload = fill_workload.replace('workers="2"', f'workers="{workers}"')
```

Comment:
@hmaheswa worker count is updating in all places. It should update only for prepare work stage and for init work stage, provide worker count as 1

Reply:
done @udaysk23